packages/@lwc/engine-dom/src/custom-elements/create-custom-element.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { setPrototypeOf } from '@lwc/shared';
+import { createPivotConstructor } from './create-pivot-constructor';
+import { hasCustomElements } from './has-custom-elements';
+
+export type LifecycleCallback = (elm: HTMLElement) => void;
+export interface UpgradableCustomElementConstructor extends CustomElementConstructor {
+    new (upgradeCallback: LifecycleCallback): HTMLElement;
+}
+
+export let createCustomElement: (
+    tagName: string,
+    upgradeCallback: LifecycleCallback,
+    connectedCallback: LifecycleCallback,
+    disconnectedCallback: LifecycleCallback
+) => HTMLElement;
+
+if (hasCustomElements) {
+    // It's important to cache window.HTMLElement here. Otherwise, someone else could overwrite window.HTMLElement (e.g.
+    // another copy of the engine, or another pivot implementation) and we would get "Illegal constructor" errors
+    // because the HTMLElement prototypes are mixed up.
+    const { HTMLElement } = window;
```

Comment:
1. How does caching the reference help with two copies of the engine? Both the engines will load the engine-dom and the second one will use the first one's patched HTMLElement correct?
2. This might be problematic from sandboxing pov. The renderer factory is executed in a sandbox. The sandbox environment might already have the HTMLElement patched before engine-dom is loaded. (_The renderer factory is executed once per sandbox and the renderer is cached_)

Reply:
When you call `myComponent extends HTMLElement`, you're grabbing whatever `HTMLElement happens to be on `window.HTMLElement` at that time. The pivot registry overwrites `window.HTMLElement`, so it's not the same one if you have multiple copies of the pivot registry.

So if you load two copies of the pivot registry in the same environment, the second one may accidentally grab the `HTMLElement` from the first one. This leads to errors at runtime that my iframe Karma tests catch. 🙂

Reply:
Added a longer comment to explain.